### PR TITLE
[IMP] website: `s_progress_bar` revamp design

### PR DESCRIPTION
**Pull request by stefanorigano**
This PR updates the progress bar component, improving its design and slightly expanding functionality. The visual result now align with the new design line, and new customization options have been introduced. The labels can be placed at the bottom or side of the bar, or removed entirely. The snippets now adapts its background color to adapt to the parent one, replacing the previous grey background.

Accessibility has been enhanced by updating the DOM structure to match Bootstrap 5.3 migration guidelines[1]. To prevent layout issues, overflow-visible has been implemented and a minimum height has been set, ensuring labels are readable and never truncated.
Auto-preview has been removed from some widgets to reduce visual clutter.

[1] https://getbootstrap.com/docs/5.3/migration/#progress-bars

task-4070098
part of task-3619705

requires **(?)**:
- https://github.com/odoo/design-themes/pull/841

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by stefanorigano** [comment]
**_Note for the reviewer_**
design-themes is currently red 'cause of a "versioning issue"... but the snippet it's neither called or customized by themes. Unfortunately I was not able to address the issue.

I do suspect that the issue is caused by legacy code or by exceptions at the the python test itself.

```
FAIL: TestNewPageTemplates.test_render_applied_templates
Traceback (most recent call last):
  File "/data/build/design-themes/test_themes/tests/test_new_page_templates.py", line 209, in test_render_applied_templates
    self.assertFalse(errors, "No error should have been collected")
AssertionError: ["Using 'no theme', view 'website.snippets' contains conflicting classes: {'s_progress_bar_label_inline', 's_progress_bar'} [...]
```

File: addons/website/static/src/snippets/s_progress_bar/options.js [hunk 1]
```diff
@@ -25,16 +24,25 @@ options.registry.progress = options.Class.extend({
             this.$target.find('.progress-bar span').addClass('s_progress_bar_text');
         }
 
-        let $text = this.$target.find('.s_progress_bar_text');
-        if (!$text.length) {
-            $text = $('<span/>').addClass('s_progress_bar_text').html(_t('80% Development'));
+        const progress = this.$target[0].querySelector(".progress");
+        const progressValue = progress.getAttribute('aria-valuenow');
```
Comment (bso-odoo) [inline addons/website/static/src/snippets/s_progress_bar/options.js:28]: Let's use double-quotes around all strings.

**Review by bso-odoo — CHANGES_REQUESTED** [review]
Something is wrong, I reach the following display using the branch - and also after a rebase:
image
Scenario:
- drop text block
- drop progress bar x3 + set different label position in each

**Comment by stefanorigano** [comment]
@bso-odoo , `vcss="000"` -> `data-vcss="000"` .. sorry for the hassle 🤦🏻‍♂️
Hopefully it will solve also the [versioning issues](https://github.com/odoo/odoo/pull/174286#issuecomment-2247347565) I had. 

- used double quotes for JS strings
- BS animation fix moved in `bootstrap_review.scss`(it's needed by both backend and frontend) + and in a separated commit 
-branch rebased

Thanks

**Comment by bso-odoo** [comment]
Strange the display works for some time, but after editing, adding progress bars and saving a few times, the progress bars end up with `.progress` having no width defined - leading 21px.
I suspect this might be due to the version of the old CSS (in default theme) already being 000... maybe this clashes with the new 000 and cause problem. Let's try naming the new one 001 instead.

**Comment by Brieuc-brd** [comment]
Hello @bso-odoo 👋 
I've updated the branches 🟢 🙂 
Could you take a look ?

The diffs :
- https://github.com/odoo/odoo/compare/859dd8d304cba46fa85f2b797a9d343732f1cee7..5d854452b78aed3cd287285fbfc7bb7fe8680d1e
- https://github.com/odoo/odoo/compare/ac8605e2d12e442665bc86bf4f4bd8d41b8f4cfe..abc601f5a41f13c361684a67d15dc07f375fa14f
- https://github.com/odoo/design-themes/compare/f88697ce25c8015c183bc258e2f94180646461c7..3e04bbd61e0cccaf31c54fd2411e37ea15104339

**Comment by bso-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**